### PR TITLE
Let the uniformity analysis trust the handle validation pass.

### PR DESCRIPTION
**Pull request by jimblandy**
Undo some changes from #1668, now that #2090 has been merged.

cc @ErichDonGubler

### CODE REVIEW

**Review by ErichDonGubler — COMMENTED** [review]
LGTM so far, after skimming through everything. Wanted to double-check what cases still remain on my desktop in my tomorrow morning.

File: src/valid/analyzer.rs [hunk 1]
```diff
@@ -689,13 +685,7 @@ impl FunctionInfo {
                 non_uniform_result: self.add_ref(expr),
                 requirements: UniformityRequirements::empty(),
             },
-            E::CallResult(function) => {
-                let info = other_functions
-                    .get(function.index())
-                    .ok_or(ExpressionError::CallToUndeclaredFunction(function))?;
```
Comment (teoxoy) [inline src/valid/analyzer.rs:695]: We can also remove `ExpressionError::CallToUndeclaredFunction` since this was the only place it was used.

**Review by teoxoy — APPROVED** [review]
LGTM, besides the comment

**Comment by jimblandy** [comment]
Rebase after a bad merge.

**Pull request merged**